File: services/api/internal/domain/meeting.go
package domain

type MeetingStatus string

const (
	MeetingStatusScheduled MeetingStatus = "scheduled"
	MeetingStatusLive      MeetingStatus = "live"
	MeetingStatusEnded     MeetingStatus = "ended"
)

type Meeting struct {
	ID         string        `json:"id"`
	Title      string        `json:"title"`
	HostUserID string        `json:"hostUserId"`
	Status     MeetingStatus `json:"status"`
	RoomName   string        `json:"roomName"`
	StartTime  string        `json:"startTime,omitempty"`
	EndTime    string        `json:"endTime,omitempty"`
}

type Todo struct {
	Owner   string `json:"owner"`
	Content string `json:"content"`
	Due     string `json:"due"`
}

type MeetingSummary struct {
	MeetingID  string `json:"meetingId"`
	Summary    string `json:"summary"`
	Todos      []Todo `json:"todos"`
	StorageURI string `json:"storageUri"`
}

File: packages/contracts/meeting.ts
export type MeetingId = string;
export type MeetingStatus = "scheduled" | "live" | "ended";
export type JobStatus = "queued" | "processing" | "done" | "failed";

export interface MeetingSummary {
  meetingId: MeetingId;
  title: string;
  status: MeetingStatus;
  roomName: string;
  storageUri?: string;
}

export interface RecordingEvent {
  event: "egress_ended";
  roomName: string;
}

export interface PostMeetingJob {
  jobId: string;
  meetingId: MeetingId;
  status: JobStatus;
  stage: "recording" | "transcription" | "summary" | "storage" | "index";
}
